Extension/src/Debugger/configurationProvider.ts

```diff
@@ -231,11 +231,61 @@ class CppConfigurationProvider implements vscode.DebugConfigurationProvider {
                     config.pipeTransport.pipeProgram = replacedPipeProgram;
                 }
             }
+
+            const macOSMIMode: string = config.osx?.MIMode ?? config.MIMode;
+            const macOSMIDebuggerPath: string = config.osx?.miDebuggerPath ?? config.miDebuggerPath;
+
+            // Validate LLDB-MI
+            if (os.platform() === 'darwin' && // Check for macOS
+                (!macOSMIMode || macOSMIMode === 'lldb') &&
+                !macOSMIDebuggerPath && // User did not provide custom lldb-mi
+                !this.hasLLDBFramework()
+                ) {
+
+                const moreInfoMessage: string = localize("lldb.framework.more.info", "More Info");
+                const LLDBFrameworkMissingMessage: string = localize("ldlb.framework.not.found", "Missing 'LLDB.framework' for lldb-mi.");
+
+                vscode.window.showErrorMessage(LLDBFrameworkMissingMessage, moreInfoMessage)
+                 .then(value => {
+                    if (value === moreInfoMessage) {
+                        let helpURL: string = "https://aka.ms/VSCode-CPP-LLDBFrameworkNotFoundHelp";
+                        vscode.env.openExternal(vscode.Uri.parse(helpURL));
+                    }
+                });
+
+                return undefined;
+            }
         }
         // if config or type is not specified, return null to trigger VS Code to open a configuration file https://github.com/Microsoft/vscode/issues/54213
         return config && config.type ? config : null;
     }
 
+    private hasLLDBFramework(): boolean {
+        const LLDBFramework: string = "LLDB.framework";
+        let searchPaths: string[] = [
+            "/Library/Developer/CommandLineTools/Library/PrivateFrameworks", // XCode CLI
+            "/Applications/Xcode.app/Contents/SharedFrameworks" // App Store XCode
+        ];
```

Comment:
What if the user has it next to lldb-mi in the extensions folder or maps it? should we also check there just in case?

Reply:
Then they would point to their own custom `lldb-mi`.